docs/design/2020-01-24-collations.md

```diff
@@ -0,0 +1,294 @@
+  * If TiDB supports all the collations, all of their implementations should be analyzed carefully. That is a lot of work.
+
+### Compatibility between TiDB versions
+
+#### Requirements
+
+In this proposal, both of the compatibility issues can be solved in the new version of TiDB, and with the help of the version of metadata in TiDB, we can always indicate if a collation is old(the "fake one") or new(the "real one"). However, here we propose several requirements below and careful considerations are needed to meet them as many as possible:
+
+  1. For an existing TiDB cluster, its behavior remains unchanged after upgrading to newer version
+    - a. After upgrade, the behavior of existing tables remains unchanged.
+    - b. After upgrade, the behavior of newly-created tables remains unchanged.
+    - c. If a TiDB cluster is replicating the data to a MySQL instance through Binlog, all  replications remain unchanged.
+  2. Users of MySQL who requires collations can move to TiDB without modifying their scripts/applications.
+  3. Users of TiDB can distinguish the tables with old collation and new collations easily, if they both exist in a TiDB cluster.
+  4. All replications/backups/recovers through TiDB tools like Binlog-related mechanism, CDC and BR should not encounter unexpected errors caused from the changes of collations in TiDB. For example, 
+    - When using BR, if backup data from old cluster is applied to the new cluster, the new cluster should know that the data is binary collated without padding, or otherwise 'Duplicate entry' error may be reported for primary key/unique key columns.
+    - If we don't allow both old collations and new collations exist in a TiDB cluster(the Option 4, see below), trying to make replications/backup & recovery between old/new clusters will break this prerequisite.
+
+#### Option 1
+
+Add a series of new collatins named with the suffix "`_np_bin`"(meaning "NO PADDING BINARY"), for example, `utf8mb4_np_bin`. Such new collations don't exist in MySQL, but they're the "real ones" used by current TiDB. After upgrading to newer TiDB versions, all old collations are shown as "`_np_bin`", MySQL collations behave the same with MySQL.
+
+Pros: Requirement 1.a, 2, 3 and 4 are met.
+
+Cons: Requirement 1.b, 1.c are not met.
+
+#### Option 2
+
+Keep all collations defined in TiDB as what they were, define a series of new collations that are actually compatible with MySQL collations. For example, we can create a new `tidb_utf8_mb4_general_ci` that is the same as `utf8mb4_general_ci` in MySQL. When TiDB users want the "real" collations, they can modify their `CREATE TABLE` statements to use the new ones.
+
+Pros: Requirement 1.a, 1.b, 2 and 3 are met.
+
+Cons: Requirement 1.c and 4 are not met, the namings are also confusing: collations that are named from MySQL are different from MySQL, but the ones named different from MySQL behave the same as MySQL.
```

Comment:
Requirement 2 is not met, since new users need to add `tidb_` prefix to actually use specified collations.

Reply:
Got it, thanks.